src/common/subscription/SignupForm.ts

```diff
@@ -35,6 +35,8 @@ import { SubscriptionApp } from "./SubscriptionViewer"
 export type SignupFormAttrs = {
 	/** Handle a new account signup. if readonly then the argument will always be null */
 	onComplete: (arg0: NewAccountData | null) => void
+	/** Handle failed signup, in case of failed captcha for example */
+	onFail: () => void
```

Comment:
there's an `onComplete()` above that already takes a nullable argument. Should we maybe keep one callback but improvement the argument type? something like

```ts
onComplete: (singupResult: { type: "success", data: NewAccountData } | { type: "failure" }) => unknown
```

or we keep it like it is but then `onComplete()` should take non-nullable argument if I understand it right